feat(Pageable): GlobalPageable 처리

yawning5 opened the pull request:
Controller 에서 Pageable 객체로 입력을 받는 Controller 에 별도의 처리를 하지않아도 적용이 됩니다

임시 Controller 테스트 패키지에 넣고 Controller 적용 테스트 완료 하였습니다

dnjsals45 committed:
chore(ci): CI 파이프라인 작성

dnjsals45 committed:
chore(ci): build 실행 name 변경(Run test -> Run build)

yawning5 committed:
fix(order): findAccessibleOrder else 제거

yawning5 committed:
feat(page): global 설정에 pageable 관련 검증 및 제한 설정 추가

yawning5 committed:
Merge branch 'feat/order/statusChange' into feat/global/pageConfig

# Conflicts:
#	src/main/java/com/sparta/tdd/domain/order/service/OrderService.java

yawning5 committed:
feat(page): global 설정에 pageable 관련 검증 및 제한 설정 추가

yawning5 committed:
Merge branch 'dev' into feat/global/pageConfig

dnjsals45 committed:
chore(common): ai관련 api key를 application.yml에 등록

dnjsals45 committed:
refactor(common): JPA 메서드가 올바르게 수행되도록 메서드명 변경

dnjsals45 committed:
refactor(common): 데이터 정리 작업을 @BeforeEach 대신 @AfterEach로 변경해 명확한 작업 의도 표현

dnjsals45 committed:
refactor(common): 통과되지 않는 테스트 코드 수정

dnjsals45 committed:
refactor(common): CI 환경에서 로그가 나오지 않도록 설정

dnjsals45 committed:
refactor(common): Repository 테스트에 모두 템플릿 적용

dnjsals45 committed:
chore(common): CI 파이프라인에서 ddl 로그 숨기도록 설정

dnjsals45 committed:
chore(common): 로그 레벨 제어가 아닌 표준 출력을 리다이렉션하여 DDL 문구 제거

yawning5 committed:
feat(Page): Pageable 공통처리

yawning5 committed:
fix(Review): WebConfig 에 PageableCustom 을 적용하여서 ControllerTest 시에 클래스단에 @EnableSpringDataWebSupport 를 붙여 SpringMVC 웹 지원을 받게끔 설정해주시면 되겠습니다

yawning5 committed:
Merge branch 'dev' into feat/global/pageConfig

dnjsals45 commented on src/main/java/com/sparta/tdd/global/pageable/CustomPageableResolver.java:
```diff
@@ -0,0 +1,74 @@
+        int capped = Math.min(base.getPageSize(), MAX_SIZE);
+        int normalized = ALLOWED.contains(capped)
+            ? capped
+            : DEFAULT_PAGE_SIZE;
```
입력받은 pageSize와 MAX_SIZE만 비교하게 된다면, 50 이상의 값이 들어오게 되었을 때 10이 아니라 50으로 나오게 되는 문제가 발생하게 될 것 같습니다

예를 들어 100이 들어올 경우, capped의 값은 50이 되고 그렇다면 normalized된 값은 10이 나와야 하는데 50은 ALLOWED에 포함되어 있어 50의 값으로 정해지게 될 것 같아요

yawning5 replied:
수정하였습니다 감사합니다!

dnjsals45 commented on src/main/java/com/sparta/tdd/global/config/WebConfig.java:
```diff
@@ -0,0 +1,40 @@
+        return new CustomPageableResolver(sortHandlerMethodArgumentResolver);
+    }
+
+    @Override
+    public void addArgumentResolvers(List<HandlerMethodArgumentResolver> resolvers) {
+        resolvers.add(sortHandlerMethodArgumentResolver());
+    }
```
현재 메인 코드에서는 resolver에 `sortHandlerMethodArgumentResolver()`가 등록되고 있습니다.

하지만 테스트 코드에서는
```java
// 2) 커스텀 Pageable 리졸버
var pageableResolver = new CustomPageableResolver(sortResolver);

// 3) 컨트롤러 인스턴스 직접 생성 (의존성 있으면 Mockito로 목 생성자 주입)
var controller = new PageProbeController();

// 4) Standalone MockMvc 구성
mvc = MockMvcBuilders
.standaloneSetup(controller)
.setCustomArgumentResolvers(pageableResolver /*, sortResolver - Sort를 직접 파라미터로 받는 엔드포인트가 있을 때만 */)
```
와 같이 CustomPageableResolver를 등록하고 있어요.

실제 swagger 테스트를 진행해보면 단순 `sortHandlerMethodArgumentResolver()` 등록만으로는 pageable 인터페이스의 구현체를 찾을 수 없고, pageable 인터페이스를 인스턴스화하려고 하기 떄문에 오류가 발생하게 됩니다.

yawning5 replied:
확인 감사합니다 추후 시간되시면 swagger 관련해서 여쭤봐도 될까요?
수정완료하였습니다

yawning5 committed:
fix(Pageable):

 WebConfig.addArgumentResolvers 에서 CustomPageableResolver 가 아닌 sortHandler 만 add 하고 있던 문제 해결

 CustomPageableResolver 에서 50 이상의 값이 들어올시 10이 아닌 50으로 고정되었던 문제 해결

p990805 reviewed (commented):
고생하셨습니다. ⚡

yawning5 committed:
fix(Pageable):

@Nullable 적용위치 변경

yawning5 committed:
fix(Order): OrderService.findAccessibleOrder 예외코드 변경

dnjsals45 reviewed (approved):
고생하셨습니다!
